ansible/roles/debops.tinyproxy/defaults/main.yml

```diff
@@ -0,0 +1,501 @@
+          Syslog: Tell tinyproxy to use syslog instead of a logfile.  This
+          option must not be enabled if the Logfile directive is being used.
+          These two directives are mutually exclusive.
+        raw: |
+          Syslog On
+        state: 'comment'
+
+      - name: 'LogLevel'
+        comment: |
+          LogLevel:
+
+          Set the logging level. Allowed settings are:
+          Critical  (least verbose)
+          Error
+          Warning
+          Notice
+          Connect   (to log connections without Info's noise)
+          Info    (most verbose)
+
+          The LogLevel logs from the set level and above. For example, if the
+          LogLevel was set to Warning, then all log messages from Warning to
+          Critical would be output, but Notice and below would be suppressed.
+        raw: |
+          LogLevel Info
+        state: 'present'
+
+      - name: 'PidFile'
+        comment: |
+          PidFile: Write the PID of the main tinyproxy thread to this file so it
+          can be used for signalling purposes.
+        raw: |
+          PidFile "/run/tinyproxy/tinyproxy.pid"
+        state: 'present'
+
+      - name: 'XTinyproxy'
+        comment: |
+          XTinyproxy: Tell Tinyproxy to include the X-Tinyproxy header, which
+          contains the client's IP address.
+        raw: |
+          XTinyproxy Yes
+        state: 'comment'
+
+      - name: 'upstream'
+        comment: |
+          Upstream:
+
+          Turns on upstream proxy support.
+
+          The upstream rules allow you to selectively route upstream connections
+          based on the host/domain of the site being accessed.
+
+          For example:
+           # connection to test domain goes through testproxy
+           upstream testproxy:8008 ".test.domain.invalid"
+           upstream testproxy:8008 ".our_testbed.example.com"
+           upstream testproxy:8008 "192.168.128.0/255.255.254.0"
+
+           # no upstream proxy for internal websites and unqualified hosts
+           no upstream ".internal.example.com"
+           no upstream "www.example.com"
+           no upstream "10.0.0.0/8"
+           no upstream "192.168.0.0/255.255.254.0"
+           no upstream "."
+
+           # connection to these boxes go through their DMZ firewalls
+           upstream cust1_firewall:8008 "testbed_for_cust1"
+           upstream cust2_firewall:8008 "testbed_for_cust2"
+
+           # default upstream is internet firewall
+           upstream firewall.internal.example.com:80
+
+          The LAST matching rule wins the route decision.  As you can see, you
+          can use a host, or a domain:
+           name     matches host exactly
+           .name    matches any host in domain "name"
+           .        matches any host with no domain (in 'empty' domain)
+           IP/bits  matches network/mask
+           IP/mask  matches network/mask
+        raw: |
+          Upstream some.remote.proxy:port
+        state: 'comment'
```

Comment:
Since upstream definition will be the first go-to to configure after the role is applied, this section could be defined in separate set of configuration variables, for example `tinyproxy__*_upstreams`. Since the service uses just one configuration file, you would have to duplicate the templating code, but this would allow more flexibility in defining the upstreams.

Also, the above configuration seems to be from an older Tinyproxy version. I'm comparing this with Tinyproxy 1.10 included in Debian Buster and there the upstream configuration is a bit more extensive, with support for http and socks connections. What Tinyproxy version did you use as a base?